From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Thu, 7 Jan 2021 15:14:40 +0100
Subject: [PATCH] metno: Use compat 2.0 API

Version 1.9, which we were using, was deprecated and will be removed soon,
so migrate to the compat XML output of the v2.0 API.

Closes: #65
---
 libgweather/weather-yrno.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/libgweather/weather-yrno.c b/libgweather/weather-yrno.c
index 8324121f..f41cbe68 100644
--- a/libgweather/weather-yrno.c
+++ b/libgweather/weather-yrno.c
@@ -37,7 +37,7 @@
 
 #define XC(t) ((const xmlChar *)(t))
 
-/* Reference for symbols at http://om.yr.no/forklaring/symbol/ */
+/* Reference for symbols at https://api.met.no/weatherapi/weathericon/2.0/ */
 typedef struct {
     int code;
     GWeatherSky sky;
@@ -443,12 +443,12 @@ yrno_start_open (GWeatherInfo *info)
     if (!loc->latlon_valid)
 	return FALSE;
 
-    /* see the description here: https://api.met.no/ */
+    /* see the description here: https://api.met.no/weatherapi/locationforecast/2.0/documentation */
 
     g_ascii_dtostr (latstr, sizeof(latstr), RADIANS_TO_DEGREES (loc->latitude));
     g_ascii_dtostr (lonstr, sizeof(lonstr), RADIANS_TO_DEGREES (loc->longitude));
 
-    url = g_strdup_printf("https://api.met.no/weatherapi/locationforecast/1.9/?lat=%s;lon=%s", latstr, lonstr);
+    url = g_strdup_printf("https://api.met.no/weatherapi/locationforecast/2.0/classic?lat=%s;lon=%s", latstr, lonstr);
     g_debug ("yrno_start_open, requesting: %s", url);
 
     message = soup_message_new ("GET", url);
